### PR TITLE
Remove UserController

### PR DESCRIPTION
**Pull request by vpereira**
To accomplish it we did:

- Moved home action to WebuiController and adapted the routes
- Removed the old and unused routes '/home/my_work' and '/home/my_project'
- Removed the UserController

Note:

Since nobody access the following routes, and they were already marked as "old routes" since 2.9, I dropped them. Should I add a note to the release notes?

The routes: 

```
  get 'home/my_work' => :home
  get 'home/list_my' => :home
  get 'home/home_project' => :home_project
```

For me it would be ok  to remove the '/home' endpoint too and instead use the `/users/$USER` directly. What do you think @hennevogel ? I deprecated it,  removing it from the partials `_logged_in_user_navigation.html.haml` and `obs_factory/_personal_navigation.html.haml` but it still accessible.

### CODE REVIEW

**Comment by coolo** [comment]
I can tell you that home/my_work used to be my 'browser homepage' - but I assume most people don't care as much :)

**Comment by vpereira** [comment]
> I can tell you that home/my_work used to be my 'browser homepage' - but I assume most people don't care as much :)

Looking the logs, just bots are crawling it. Today we have the "My Tasks" which probably makes more sense that this "my_work".

**Comment by dmarcoux** [comment]
I would add a note in the release notes about the deleted routes.

File: src/api/app/views/layouts/webui/_logged_in_user_navigation.html.haml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 %li.nav-item
-  = link_to(home_path, id: 'link-to-user-home', class: 'nav-link') do
+  = link_to(user_path(User.session!.login), id: 'link-to-user-home', class: 'nav-link') do
```
Comment (dmarcoux) [inline src/api/app/views/layouts/webui/_logged_in_user_navigation.html.haml]: `.login` isn't needed.

Comment (vpereira) [reply src/api/app/views/layouts/webui/_logged_in_user_navigation.html.haml]: done

**Comment by vpereira** [comment]
> I would add a note in the release notes about the deleted routes.

done!

**Pull request merged**